include/nccl_ofi_rdma.h

```diff
@@ -88,22 +111,34 @@ typedef struct nccl_net_ofi_rdma_mr_handle {
    destination buffer */
 typedef struct nccl_net_ofi_rdma_ctrl_msg {
 	/* Message type, must be NCCL_OFI_RDMA_MSG_CTRL */
-	nccl_ofi_rdma_msg_type_t type;
+	uint32_t type:NCCL_OFI_RDMA_CTRL_TYPE_BITS;
 
 	/* Message sequence number */
-	uint16_t msg_seq_num;
+	uint32_t msg_seq_num:NCCL_OFI_RDMA_SEQ_BITS;
 
 	/* A comm identitifer that uniquely identifies the comm
 	 * on the receiver side */
-	uint32_t remote_comm_id;
+	uint32_t remote_comm_id:NCCL_OFI_RDMA_COMM_ID_BITS;
+
+	uint32_t buff_len;
 
 	uint64_t buff_addr;
-	uint64_t buff_len;
-	uint64_t buff_mr_key[MAX_NUM_RAILS];
+
+	union {
+		uint32_t short_buff_mr_key[MAX_NUM_RAILS];
+		uint64_t long_buff_mr_key[MAX_NUM_RAILS];
+	};
```

Comment:
this is not getting to 32B yet, so I am guessing this is an initial step. Do you have a plan to change it later?

Reply:
When using short MR keys, the control message is of 32B exactly. See how `nccl_net_ofi_rdma_ctrl_msg_size()` will return 32B in that case.

Reply:
oh I see, even if the struct is larger, we are not sending it all. Thanks!